### PR TITLE
GP2-1887 GP2-2136 GP2-2171 VFM questionnaire

### PR DESCRIPTION
**Pull request by bobmeredith**
This changeset implements the VFM survey - apart from company house lookup.
Questions are defined in a table in directory-sso and users' answers are stored in a corresponding table linked to the question.
The complete questionnaire is retrieved along with a user's answers by an ajax call made from the questionnaire react component. The FE then calculates the position in the stream of questions and presents the correct question to the user.
The questionnaire uses a new - re-usable modal component.
The questionnaire is only trigggered by adding the query parameter '?startsurvey' to the url on the dashboard page.  On triggered, the questionnaire won't let go until completed.

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-1887
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [x] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] I have updated security dependencies
- [ ] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Comment by bobmeredith** [comment]
image

image

image

**Comment by bobmeredith** [comment]
Testing - 
Ensure questions are loaded in directory-sso (make manage loaddata sso/user/fixtures/*.json)
Go to dashboard and add '?startsurvey' to the end of the url.  this will trigger the survey.
Answer questions. refresh page halfway through to check recovery and back button.
If bottom answers selected from first questions, survey ends - otherwise will go to end.
On completion, check survey does not reappear - even if trigger parameter applied.
Questions and answers can be manipulated in directory-sso admin.

File: core/mixins.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ class EnableSegmentationMixin:
     # used by the page modal manager React component
     def get_context(self, request):
         context = super().get_context(request)
+        if request.user.is_authenticated and 'startsurvey' in request.GET:
+            request.user.set_user_questionnaire_answer(0, 'in-progress')
         if request.user.is_authenticated and request.user.user_profile:
```
Comment (rafa-garcia) [inline core/mixins.py:56]: Do we need a `segment` in the context anymore?

**Review by rafa-garcia — APPROVED** [review]
Leaving to you whether to keep or remove the segment from the context and the tour mixin. LGTM

**Pull request merged**